### PR TITLE
Update altair to 5.4.1

### PR DESCRIPTION
**Pull request by joelostblom**
### Description

Update altair to 5.4.1, depends on https://github.com/pyodide/pyodide/pull/5121 since our dependencies changed in 5.4.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry

### CODE REVIEW

**Review by agriyakhetarpal — APPROVED** [review]
Thanks, @joelostblom!

File: packages/altair/meta.yaml
Comment (ryanking13) [inline packages/altair/meta.yaml:23]: I think this PR would need to wait until #5121 is merged.

Comment (mattijn) [reply packages/altair/meta.yaml:23]: #5121 is merged. Can we re-run the failing CI?🙏

Comment (joelostblom) [reply packages/altair/meta.yaml:23]: I retriggered by merging in the main branch after fixing a conflict, but it seems like the macos14 selenium test is still failing for what seems to be an unrelated reason:

image

**Comment by hoodmane** [comment]
Don't worry about test-core-safari or test-core-node. But the actual test is failing, I think it's because the updated version of altair dropped the dependency on pandas?
```patch
--- a/packages/altair/test_altair.py
+++ b/packages/altair/test_altair.py
@@ -3,7 +3,7 @@ from pytest_pyodide import run_in_pyodide
 
 
 @pytest.mark.xfail_browsers(firefox="slow")
-@run_in_pyodide(packages=["altair"])
+@run_in_pyodide(packages=["altair", "pandas"])
 def test_altair(selenium):
     import altair as alt
     import pandas as pd
```

File: packages/altair/meta.yaml
Comment (ryanking13) [reply packages/altair/meta.yaml:23]: It is an upstream selenium issue, so please ignore it.

**Comment by joelostblom** [comment]
Thanks you @hoodmane , I included your suggested change in [628d64c](https://github.com/pyodide/pyodide/pull/5124/commits/628d64c73338b7904a32dac69bf39802431074e6)

File: packages/altair/test_altair.py
Comment (agriyakhetarpal) [inline packages/altair/test_altair.py:9]: This can fail in some cases (as it did in this instance) because `pandas` was removed as a runtime dependency above, so there is a chance `pandas` will not have been built by then (or will have been in the process of building in a different CI job). I am not sure if there is a reliable way to test both packages without one depending on another, do you have any thoughts, @ryanking13?

Comment (joelostblom) [reply packages/altair/test_altair.py]: Thanks for that info! Technically, we don't need to use pandas and can create chart data via altair directly. I switched to using one of these approaches since it seems like an easier way to get this test to work.

**Review by agriyakhetarpal — APPROVED** [review]
Thanks, @joelostblom!

File: packages/altair/test_altair.py
Comment (ryanking13) [reply packages/altair/test_altair.py]: You can use a similar approach [we use in duckdb test](https://github.com/pyodide/pyodide/blob/a5386c86b86135bda9c6f5d2d4f214726bacb7af/packages/duckdb/test_duckdb.py#L16); skipping the test until the dependency is built. But updating the test sounds better.

**Pull request merged**
